### PR TITLE
Add support for the ESP32-P4

### PR DESCRIPTION
**Pull request by tannewt**
USB doesn't currently work. TinyUSB changes need to be upstreamed.

### CODE REVIEW

File: ports/espressif/peripherals/esp32p4/pins.c [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+//
+// SPDX-FileCopyrightText: Copyright (c) 2021 microDev
+//
+// SPDX-License-Identifier: MIT
+
+#include "peripherals/pins.h"
+
+const mcu_pin_obj_t pin_GPIO0 = PIN(0, NO_ADC, NO_ADC_CHANNEL, NO_TOUCH_CHANNEL);
+const mcu_pin_obj_t pin_GPIO1 = PIN(1, ADC_UNIT_1, ADC_CHANNEL_0, TOUCH_PAD_NUM1);
+const mcu_pin_obj_t pin_GPIO2 = PIN(2, ADC_UNIT_1, ADC_CHANNEL_1, TOUCH_PAD_NUM2);
```
Comment (arturo182) [inline ports/espressif/peripherals/esp32p4/pins.c:11]: This seems to differ from the datasheet
image

File: ports/espressif/peripherals/esp32p4/pins.h [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+#define GPIO44_EXISTS 1
+extern const mcu_pin_obj_t pin_GPIO44;
+#define GPIO45_EXISTS 1
+extern const mcu_pin_obj_t pin_GPIO45;
+#define GPIO46_EXISTS 1
+extern const mcu_pin_obj_t pin_GPIO46;
+#define GPIO47_EXISTS 1
+extern const mcu_pin_obj_t pin_GPIO47;
+#define GPIO48_EXISTS 1
+extern const mcu_pin_obj_t pin_GPIO48;
```
Comment (arturo182) [inline ports/espressif/peripherals/esp32p4/pins.h:102]: 49-55 missing?

Comment (tannewt) [reply ports/espressif/peripherals/esp32p4/pins.h:102]: Added!

File: ports/espressif/peripherals/esp32p4/pins.c (hunk 1 above)
Comment (tannewt) [reply ports/espressif/peripherals/esp32p4/pins.c]: Updated. Thanks for checking!

File: ports/espressif/supervisor/port.c [hunk 3]
```diff
@@ -238,20 +247,6 @@ safe_mode_t port_init(void) {
...
-    common_hal_never_reset_pin(&pin_GPIO21);
-    #elif defined(CONFIG_IDF_TARGET_ESP32C6)
-    common_hal_never_reset_pin(&pin_GPIO16);
-    common_hal_never_reset_pin(&pin_GPIO17);
-    #elif defined(CONFIG_IDF_TARGET_ESP32S2) || defined(CONFIG_IDF_TARGET_ESP32S3)
-    common_hal_never_reset_pin(&pin_GPIO43);
-    common_hal_never_reset_pin(&pin_GPIO44);
-    #endif
-    #endif
-
```
Comment (dhalbert) [inline ports/espressif/supervisor/port.c:254]: Are these now moot never-resets due to something else?

**Review by dhalbert — CHANGES_REQUESTED** [review]
Just one q

File: ports/espressif/supervisor/port.c (hunk 3 above)
Comment (tannewt) [reply ports/espressif/supervisor/port.c:254]: Yup! The reset forbidden stuff covers it: https://github.com/adafruit/circuitpython/blob/main/ports/espressif/common-hal/microcontroller/Pin.c#L122

**Review by dhalbert — APPROVED** [review]
Thanks for the quick work and thanks to @bwhitman also!

**Pull request merged**